File: SWEA/D2/1959. 두 개의 숫자열/두 개의 숫자열.py
t = int(input())

for test in range(1, t + 1):
    n, m = map(int, input().split())

    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

    if len(a) > len(b):
        result = 0
        for i in range(len(a) - len(b) + 1):
            temp = 0
            for j in range(len(b)):
                temp += a[i + j] * b[j]
            if temp > result:
                result = temp
    else:
        result = 0
        for i in range(len(b) - len(a) + 1):
            temp = 0
            for j in range(len(a)):
                temp += b[i + j] * a[j]
            if temp > result:
                result = temp

    print(f"#{test} ", end='')
    print(result)

File: SWEA/D2/1959. 두 개의 숫자열/두 개의 숫자열.java
//import java.io.FileInputStream;
import java.util.Scanner;

public class Solution {

	public static void main(String[] args) throws Exception {
//		System.setIn(new FileInputStream("src/input.txt"));
		Scanner sc = new Scanner(System.in);
		
		int t = sc.nextInt();
		for (int tc = 1; tc <= t; tc++) {
			int n = sc.nextInt();
			int m = sc.nextInt();
			
			int[] a = new int[n];
			int[] b = new int[m];
			for (int i = 0; i < n; i++) {
				a[i] = sc.nextInt();
			}
			for (int i = 0; i < m; i++) {
				b[i] = sc.nextInt();
			}
			
			int result = Integer.MIN_VALUE;
			for (int i = 0; i < Math.abs(m - n) + 1; i++) {
				int temp = 0;
				for (int j = 0; j < Math.min(n, m); j++) {
					if (n < m) 
						temp += a[j] * b[i + j];
					else
						temp += a[i + j] * b[j];
				}
				if (temp > result)
					result = temp;
			}
			
			System.out.printf("#%d %d%n", tc,  result);
		}

	}

}